Guides Build Status / docs job does not catch warnings as errors

If you look at pull request #8106 you'll see "all checks have passed" including the one for docs:

![Screen Shot 2021-09-28 at 11 17 46 AM](https://user-images.githubusercontent.com/21006/135116109-884f60a3-adb7-4078-bbff-c3f2fa2bbe23.png)

However, if you get on the branch locally and run `make html` the build fails:

![Screen Shot 2021-09-28 at 11 18 54 AM](https://user-images.githubusercontent.com/21006/135116320-d8b366c3-2518-4ea3-9bb1-848c8ae36441.png)

This means that the GitHub action for docs is giving us a false sense of security. It's saying the docs are ok, but they won't actually build.

Interestingly, we do see the warnings in the pull request:

![Screen Shot 2021-09-28 at 11 15 46 AM](https://user-images.githubusercontent.com/21006/135116111-fd965d2d-bb2f-4b47-9fce-61a03eea5388.png)

So maybe we need to treat the warnings as errors in the docs GitHub Action, like we do in the Makefile? This is with `SPHINXOPTS    = -W`

```
(venv) HMDC-beamish:sphinx-guides pdurbin$ sphinx-build --help | grep W -B6
  Console output options:
    -v                increase verbosity (can be repeated)
    -q                no output on stdout, just warnings on stderr
    -Q                no output at all, not even warnings
    -N                do not emit colored output
    -w FILE           write warnings (and errors) to given file
    -W                turn warnings into errors
```

I upgraded to 3.5.4 locally but I get the same error:

```
(venv) HMDC-beamish:sphinx-guides pdurbin$ make html
sphinx-build -b html -d build/doctrees  -W source build/html
Running Sphinx v3.5.4
loading translations [en]... done
loading intersphinx inventory from http://docs.python.org/objects.inv...
intersphinx inventory has moved: http://docs.python.org/objects.inv -> https://docs.python.org/3/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 100 source files that are out of date
updating environment: [new config] 100 added, 0 changed, 0 removed
reading sources... [100%] versions                                              

Warning, treated as error:
/Users/pdurbin/github/iqss/guides.dataverse.org/doc/sphinx-guides/source/api/native-api.rst:2821:Definition list ends without a blank line; unexpected unindent.
make: *** [html] Error 2
(venv) HMDC-beamish:sphinx-guides pdurbin$ git log --oneline | head -1
21fd6377d Merge branch 'develop' into 41-add-auth-to-harvest-api
(venv) HMDC-beamish:sphinx-guides pdurbin$ 
```

Please note that I'm on 21fd6377d on the 41-add-auth-to-harvest-api branch, not develop.

--- a/.github/workflows/guides_build_sphinx.yml
+++ b/.github/workflows/guides_build_sphinx.yml
@@ -9,6 +9,6 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - uses: ammaraskar/sphinx-action@master
+    - uses: OdumInstitute/sphinx-action@master
       with:
         docs-folder: "doc/sphinx-guides/"

